qualtran/cirq_interop/_cirq_interop.py

```diff
@@ -188,67 +204,97 @@ def cirq_optree_to_cbloq(
         signature = Signature([Register('qubits', 1, shape=(len(all_qubits),))])
         cirq_quregs = {'qubits': np.array(all_qubits).reshape(len(all_qubits), 1)}
 
+    cirq_quregs = {k: np.apply_along_axis(QReg, -1, v) for k, v in cirq_quregs.items()}
+
     assert signature is not None and cirq_quregs is not None
 
     bb, initial_soqs = BloqBuilder.from_signature(signature, add_registers_allowed=False)
 
-    # Magic to make sure signature of the CompositeBloq matches `Signature`.
-    qubit_to_qvar = {}
+    # 0. Helper functions
+    def _update_qreg_to_qvar(in_reg: QReg, qreg_to_qvar: Dict[QReg, Soquet]) -> None:
+        """Takes care of splits and joins to make sure `qreg_to_qvar[in_reg]` exists."""
+        if in_reg in qreg_to_qvar:
+            # This is the easy case when no split / joins are needed.
+            return
+
+        # Split everything and join s.t. joined register corresponds to `in_reg`.
```

Comment:
I suggest turning this into two comments. First: we split everything that needs splitting

Then: we join everything that needs joining

Reply:
done